clients/fossid-webapp/src/test/kotlin/FossIdClientReturnTypeTest.kt

```diff
@@ -81,84 +73,108 @@ class FossIdClientReturnTypeTest : StringSpec({
     "Scan results can be listed when there is none" {
         service.listScanResults("", "", SCAN_CODE_1) shouldNotBeNull {
             checkResponse("list scan results")
-            toList<FossIdScanResult>() should beEmpty()
+            data shouldNotBeNull {
+                this should beEmptyMap()
+            }
         }
     }
 
     "Scan results can be listed when there is some" {
         service.listScanResults("", "", SCAN_CODE_2) shouldNotBeNull {
             checkResponse("list scan results")
-            toList<FossIdScanResult>() shouldNot beEmpty()
+            data shouldNotBeNull {
+                this shouldNot beEmptyMap()
+            }
         }
     }
 
     "Identified files can be listed when there is none" {
         service.listIdentifiedFiles("", "", SCAN_CODE_1) shouldNotBeNull {
             checkResponse("list identified files")
-            toList<IdentifiedFile>() should beEmpty()
+            data shouldNotBeNull {
+                this should beEmptyMap()
+            }
         }
     }
 
     "Identified files can be listed when there is some" {
         service.listIdentifiedFiles("", "", SCAN_CODE_2) shouldNotBeNull {
             checkResponse("list identified files")
-            toList<IdentifiedFile>() shouldNot beEmpty()
+            data shouldNotBeNull {
+                this shouldNot beEmptyMap()
+            }
         }
     }
 
     "Marked as identified  files can be listed when there is none" {
         service.listMarkedAsIdentifiedFiles("", "", SCAN_CODE_1) shouldNotBeNull {
             checkResponse("list marked as identified files")
-            toList<MarkedAsIdentifiedFile>() should beEmpty()
+            data shouldNotBeNull {
+                this should beEmptyMap()
+            }
         }
     }
 
     "Marked as identified  files can be listed when there is some" {
         service.listMarkedAsIdentifiedFiles("", "", SCAN_CODE_2) shouldNotBeNull {
             checkResponse("list marked as identified files")
-            toList<MarkedAsIdentifiedFile>() shouldNot beEmpty()
+            data shouldNotBeNull {
+                this shouldNot beEmptyMap()
+            }
         }
     }
 
     "Ignored files can be listed when there is none" {
         service.listIgnoredFiles("", "", SCAN_CODE_1) shouldNotBeNull {
             checkResponse("list ignored files")
-            toList<IgnoredFile>() should beEmpty()
+            data shouldNotBeNull {
+                this should beEmpty()
+            }
         }
     }
 
     "Ignored files can be listed when there is some" {
         service.listIgnoredFiles("", "", SCAN_CODE_2) shouldNotBeNull {
             checkResponse("list ignored files")
-            toList<IgnoredFile>() shouldNot beEmpty()
+            data shouldNotBeNull {
+                this shouldNot beEmpty()
+            }
         }
     }
 
-    "Pending files can be listed when there is some" {
+    "Pending files can be listed when there is none" {
```

Comment:
Note: Ideally, the (somewhat unrelated) test case name fixes (here and below) would go to a separate / preceding commit.